src/implementation/Server/HTTPServer/actor.ts

```diff
@@ -89,8 +90,20 @@ export default class HTTPServerActor implements IServerActor {
   private async handlerDeactivate(req: IRequest, res: IResponse): Promise<IResponse> {
     const { actorTypeName, actorId } = req.params;
     const result = await ActorRuntime.getInstance(this.client.daprClient).deactivate(actorTypeName, actorId);
-    res.statusCode = HttpStatusCode.OK;
-    return this.handleResult(res, result);
+
+    switch (result) {
+      case DeactivateResult.Success:
+        res.statusCode = HttpStatusCode.OK;
+        return this.handleResult(res, result);
+      case DeactivateResult.Error:
+        res.statusCode = HttpStatusCode.INTERNAL_SERVER_ERROR;
```

Comment:
Does returning specific http status codes for the unhappy path provide value to the dapr sidecar?